Identification: BREW-133

zurek11 opened the pull request:
Resolution: Implementation

Implementation:
- Implemented MQTT error exception handling.
- Implemented endpoint for set value of module.
- Implemented DB Seed
- Added new sections to README.md (test, seed)
- Refactored models
- Implemented handling of different published MQTT messages
- Implemented Connection handler (create subscribes on server startup)
- Added flask_seeder library.

zurek11 committed:
Identification: BREW-133

Resolution: Implementation

Implementation:
- Implemented MQTT error exception handling.
- Implemented endpoint for set value of module.
- Implemented DB Seed
- Added new sections to README.md (test, seed)
- Refactored models
- Implemented handling of different published MQTT messages
- Implemented Connection handler (create subscribes on server startup)
- Added flask_seeder library.

zurek11 committed:
Identification: BREW-133

Resolution: Fix

Implementation:
- Fixed handling of received messages through MQTT.

zurek11 committed:
Identification: BREW-133

Resolution: Change

Implementation:
- Changed format of sent messages from testing devices.

tomas321 commented on core/models/module.py:
```diff
@@ -6,10 +6,10 @@
+    module_device_type_id = db.Column(UUID(as_uuid=True), db.ForeignKey('module_device_types.id'))
 
-    devices = db.relationship("Device", back_populates="module")
+    devices = db.relationship("Device", lazy='dynamic', back_populates="module")
```
good choice :+1:

tomas321 commented on mqtt/client.py:
```diff
@@ -40,34 +41,64 @@ def on_connect(self, client, userdata, flags, rc):
+
+            # Value update asynchronously
+            elif dict_message.get('values'):
+                values = dict_message.get('values')
+
+                for item in values:
+                    device = Device.query.get(item.get('id'))
+
+                    if not device:
+                        raise MQTTException(
+                            f'Špecifikovaný device nebol nájdený: {item.get("device_id")}',
```
not sure about the `id` and `device_id` keys are present in the JSON message. Please reply if I'm wrong, or else.

zurek11 replied:
Yeah you are right about it, but I suggested ( in comment) for format change: https://brewmasters.atlassian.net/wiki/spaces/TD/pages/7274550/MQTT+API

If it can't be changed to that format, I will make proper changes back.

tomas321 replied:
thx for your fast reply and clarification, i didnt notice the comment in the DOCS.. approved

tomas321 commented on mqtt/sensor.py:
```diff
@@ -33,11 +33,24 @@ def __init__(self, name, socket_io, thread_stop_event):
+                devices = module.devices
+
+                for device in devices:
+                    data['values'].append({
+                        'id': str(device.id),
+                        'value': round(random() * 10, 3),
+                        'unit': 'value',
```
similarly here, are the keys right?

zurek11 committed:
Identification: BREW-133

Resolution: Change

Implementation:
- Fixed iteration of time while saving received data to DB.

tomas321 reviewed (approved):
it's good to go
